app/pages/project/classify.jsx

```diff
@@ -0,0 +1,434 @@
+import auth from 'panoptes-client/lib/auth';
+import React from 'react';
+import PropTypes from 'prop-types';
+import createReactClass from 'create-react-class';
+import ReactDOM from 'react-dom';
+import { Helmet } from 'react-helmet';
+import apiClient from 'panoptes-client/lib/api-client';
+import counterpart from 'counterpart';
+import FinishedBanner from './finished-banner';
+import Classifier from '../../classifier';
+import seenThisSession from '../../lib/seen-this-session';
+import ClassificationQueue from '../../lib/classification-queue';
+import WorkflowAssignmentDialog from '../../components/workflow-assignment-dialog';
+import { Split } from 'seven-ten';
+
+counterpart.registerTranslations('en', {
+  classifyPage: {
+    title: 'Classify'
+  }
+});
+
+// Map each project ID to a promise of its last randomly-selected workflow ID.
+// This is to maintain the same random workflow for each project when none is specified by the user.
+const currentWorkflowForProject = {};
+
+// Map a workflow ID to a promise of its current classification resource
+// This is to maintain the same classification for each workflow.
+// In the future user might be able to specify subject sets, which we'll record here similarly.
+const currentClassifications = { forWorkflow: {}};
+
+// Queue up subjects to classify here.
+const upcomingSubjects = { forWorkflow: {}};
+
+function emptySubjectQueue() {
+  console.log('Emptying upcoming subjects queue');
+
+  for (const workflowID in upcomingSubjects.forWorkflow) {
```

Comment:
I think `Object.keys(upcomingSubjects.forWorkflow).forEach((workflowID) => {})` would preferable over `for...in` here. For loops in javascript will also iterate over non-enumerable properties the object may have inherited: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in